python/lsst/metaserv/metaREST_v0.py

```diff
@@ -0,0 +1,176 @@
+#!/usr/bin/env python
+
+# LSST Data Management System
+# Copyright 2015 AURA/LSST.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+
+"""
+This module implements the RESTful interface for Metadata Service.
+Corresponding URI: /meta. Default output format is json. Currently
+supported formats: json and html.
+
+@author  Jacek Becla, SLAC
+
+# todos:
+#  * known issue: it blocks commands such as "drop database", because
+#    it keeps connection option. Close connection per request?
+#  * migrate to db, and use execCommands etc from there.
+#  * generate proper html header
+"""
+
+from flask import Blueprint, Flask, request
+import json
+import MySQLdb
+
+from lsst.db.utils import readCredentialFile
+import lsst.log as log
+
+
+metaREST = Blueprint('metaREST', __name__, template_folder='metaserv')
+
+# connect to the metaserv database
+creds = readCredentialFile("~/.lsst/dbAuth-metaServ.txt", log)
+try:
+    con = MySQLdb.connect(host=creds['host'],
+                          port=int(creds['port']),
+                          user=creds['user'],
+                          passwd=creds['passwd'],
+                          db=creds['db'])
+
+except MySQLdb.Error as err:
+    log.info("ERROR MySQL %s", err)
+    raise
+
+def runDbQuery1(query, optTuple=None, notFoundMsg='Not found'):
+    '''Runs query that returns one row.'''
+    cursor = con.cursor()
+    if optTuple:
+        cursor.execute(query, optTuple)
+    else:
+        cursor.execute(query)
+    row = cursor.fetchone()
+    fmt = request.accept_mimetypes.best_match(['application/json', 'text/html'])
+    if not row:
+        retStr = notFoundMsg
```

Comment:
If something not found I think we are also supposed to return 404 HTTP code, not just a message string. But we should differentiate between cases when it is an error (like asking for tables in a database which does not exist) in which case we have to return error and cases when "not found" means "empty set" (and we should return OK code and empty sequence like `"{ data: [] }"` in JSON).